### PR TITLE
Select

### PR DESCRIPTION
**Pull request by Lelik7777**

### CODE REVIEW

File: mw-webapp/src/component/select/Select.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  /**
+   * Label's text
+   */
+  label: string;
+  /**
+   * Default select value (string)
+   */
+  value: string;
+  /**
+   * Select's name (string)
```
Comment (Howl404) [inline mw-webapp/src/component/select/Select.tsx:23]: Don't duplicate types in ()

File: mw-webapp/src/component/select/Select.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  /**
+   * Options list
+   */
+  options: OptionType[];
+}
+
+export const Select: React.FC<SelectProps> = (props: SelectProps) => {
+  const [selected, setSelected] = useState(props.value);
+  const handleOnChange = (e: React.ChangeEvent<HTMLSelectElement>) => setSelected(e.target.value);
+  return (
```
Comment (SergioAJS) [inline mw-webapp/src/component/select/Select.tsx:35]: Add an empty line between 34 and 35 lines

**Review by SergioAJS — CHANGES_REQUESTED** [review]

File: mw-webapp/src/component/select/Select.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import {useState} from "react";
+import styles from "src/component/select/Select.module.scss";
+
+export interface OptionType {
```
Comment (tritonJS826) [inline mw-webapp/src/component/select/Select.tsx:4]: no docs

File: mw-webapp/src/styles/_variables.scss [hunk 4]
```diff
@@ -12,4 +12,7 @@ $link-active: rgb(255, 149, 0);
 $header-background: rgb(85, 122, 149);
 
 // Title
-$title-padding: 10px;
\ No newline at end of file
+$title-padding: 10px;
+
+//Select
+$select-font-size: 1em;
```
Comment (Howl404) [inline mw-webapp/src/styles/_variables.scss:18]: I think 1em(16px) is the default font size? I don't see the point of setting it then.

File: mw-webapp/src/component/select/Select.tsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  id: string;
+  value: string;
+  text: string;
+}
+
+/**
+   * Implement a simple select component
+   */
```
Comment (tritonJS826) [inline mw-webapp/src/component/select/Select.tsx:12]: formatting problem adjust linter

File: mw-webapp/src/component/select/Select.tsx [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+export interface SelectProps {
+  /**
+   * Label's text
+   */
+  label: string;
+  /**
+   * Default select value (string)
```
Comment (tritonJS826) [inline mw-webapp/src/component/select/Select.tsx:19]: don't duplicate type in docs

File: mw-webapp/src/component/select/Select.tsx [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+   * Select's name (string)
+   */
+  name: string;
+  /**
+   * Options list
+   */
+  options: OptionType[];
+}
+
+export const Select: React.FC<SelectProps> = (props: SelectProps) => {
```
Comment (tritonJS826) [inline mw-webapp/src/component/select/Select.tsx:32]: how to work with this component? How could I add any handlers to it?

File: mw-webapp/src/styles/_variables.scss (hunk 4 above)
Comment (tritonJS826) [inline mw-webapp/src/styles/_variables.scss:18]: why in em

**Review by tritonJS826 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/component/select/Select.tsx (hunk 1 above)
Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx:23]: done

File: mw-webapp/src/component/select/Select.tsx (hunk 2 above)
Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx:35]: done

File: mw-webapp/src/styles/_variables.scss (hunk 4 above)
Comment (Lelik7777) [reply mw-webapp/src/styles/_variables.scss:18]: done

File: mw-webapp/src/component/select/Select.tsx (hunk 3 above)
Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx:4]: done

File: mw-webapp/src/component/select/Select.tsx (hunk 5 above)
Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx:12]: done

File: mw-webapp/src/component/select/Select.tsx (hunk 6 above)
Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx:19]: done

File: mw-webapp/src/styles/_variables.scss (hunk 4 above)
Comment (Lelik7777) [reply mw-webapp/src/styles/_variables.scss:18]: done

File: mw-webapp/src/component/select/Select.tsx (hunk 7 above)
Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx:34]: Did I understand correctly that I should remove useState() from the component and accept in props selected and onChange()?

Comment (Ekaterina1994) [reply mw-webapp/src/component/select/Select.tsx:34]: good idea 

Also use https://www.radix-ui.com/primitives/docs/components/select select based on radix library

File: mw-webapp/src/component/select/Select.tsx [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+export interface OptionType {
+  id: string;
+  value: string;
+  text: string;
+}
+
+// Implement a simple select component
+
+export interface SelectProps {
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/select/Select.tsx:15]: Comment should look like on strings 16-18

File: mw-webapp/src/component/select/Select.tsx [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+
+
+// Implement an option element for a select component
+
+export interface OptionType {
+  id: string;
+  value: string;
+  text: string;
+}
+
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/select/Select.tsx:12]: Add jsdoc comments to each property

**Review by Ekaterina1994 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/component/select/Select.tsx (hunk 8 above)
Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx:11]: done

File: mw-webapp/src/component/select/Select.tsx (hunk 9 above)
Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx:9]: done

File: mw-webapp/src/component/select/Select.module.scss [hunk 10]
```diff
@@ -0,0 +1,3 @@
+.select {
+cursor: pointer;
```
Comment (tritonJS826) [inline mw-webapp/src/component/select/Select.module.scss:2]: format

File: mw-webapp/src/component/select/Select.tsx [hunk 11]
```diff
@@ -0,0 +1,72 @@
+import styles from "src/component/select/Select.module.scss";
+
+/**
+ * Implement an option element for a select component
```
Comment (tritonJS826) [inline mw-webapp/src/component/select/Select.tsx:4]:
```suggestion
 * Option element for a {@link Select} component
```

File: mw-webapp/src/component/select/Select.tsx [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+ */
+export interface OptionType {
+  /**
+   * Unique value
```
Comment (tritonJS826) [inline mw-webapp/src/component/select/Select.tsx:8]:
```suggestion
   * Option id. Should be unique
```

File: mw-webapp/src/component/select/Select.tsx [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+   */
+  id: string;
+  /**
+   * Option value
+   */
+  value: string;
+  /**
+   * Option`s text
```
Comment (tritonJS826) [inline mw-webapp/src/component/select/Select.tsx:16]:
```suggestion
   * Option`s visible text
```

File: mw-webapp/src/component/select/Select.tsx [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+   */
+  text: string;
+}
+
+/**
+ * Implement a simple select component
```
Comment (tritonJS826) [inline mw-webapp/src/component/select/Select.tsx:22]:
```suggestion
 * Select component
```

File: mw-webapp/src/component/select/Select.tsx [hunk 15]
```diff
@@ -0,0 +1,72 @@
...
+   */
+  name: string;
+  /**
+   * Options list
+   */
+  options: OptionType[];
+  /**
+   * Default value for select
+   */
+  selected: string;
```
Comment (tritonJS826) [inline mw-webapp/src/component/select/Select.tsx:44]: why should we have default `value` and `selected`? Why not to use default option value or id?

File: mw-webapp/src/styles/_variables.scss [hunk 16]
```diff
@@ -12,4 +12,6 @@ $link-active: rgb(255, 149, 0);
 $header-background: rgb(85, 122, 149);
 
 // Title
-$title-padding: 10px;
\ No newline at end of file
+$title-padding: 10px;
+
```
Comment (tritonJS826) [inline mw-webapp/src/styles/_variables.scss:16]: why did you add this 2 empty lines?

File: mw-webapp/src/component/select/Select.tsx [hunk 17]
```diff
@@ -0,0 +1,72 @@
...
+    <label>
+      {props.label}
+      <select
+        name={props.name}
+        defaultValue={props.selected}
+        onChange={props.onChange}
+        className={styles.select}
+      >
+        {props.value}
+        {props.options.map((option) => (
```
Comment (tritonJS826) [inline mw-webapp/src/component/select/Select.tsx:62]: could you move our renderOption function? Btw, it could be a sub component

**Review by tritonJS826 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/component/select/Select.tsx (hunk 11 above)
Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx:4]: done

File: mw-webapp/src/component/select/Select.tsx (hunk 12 above)
Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx:8]: done

File: mw-webapp/src/component/select/Select.tsx (hunk 13 above)
Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx:16]: done

File: mw-webapp/src/component/select/Select.tsx (hunk 14 above)
Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx:22]: done

File: mw-webapp/src/component/select/Select.tsx (hunk 15 above)
Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx:44]: we can use for example options[0].value then i remove selected from props.

Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx:44]: isn`t it?

File: mw-webapp/src/styles/_variables.scss (hunk 16 above)
Comment (Lelik7777) [reply mw-webapp/src/styles/_variables.scss:16]: done

File: mw-webapp/src/component/select/Select.tsx (hunk 17 above)
Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx:62]: done

File: mw-webapp/src/component/select/Select.tsx (hunk 7 above)
Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx:51]: done

File: mw-webapp/src/component/select/Select.module.scss (hunk 10 above)
Comment (Ekaterina1994) [reply mw-webapp/src/component/select/Select.module.scss:2]: Add spaces to 2 line

File: mw-webapp/src/component/select/Select.tsx [hunk 18]
```diff
@@ -0,0 +1,74 @@
...
+
+export const Select: React.FC<SelectProps> = (props: SelectProps) => {
+  const renderOption = props.options.map((option) => (
+    <option key={option.id}
+      value={option.value}
+    >
+      {option.text}
+    </option>
+  ));
+
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/select/Select.tsx:59]: Create subfolder inside select folder with Option component

**Review by Ekaterina1994 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/component/select/Select.module.scss (hunk 10 above)
Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.module.scss:2]: done

File: mw-webapp/src/component/select/Select.tsx (hunk 15 above)
Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx:44]: done

File: mw-webapp/src/component/select/Select.tsx (hunk 18 above)
Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx:59]: done

File: mw-webapp/src/component/select/Select.tsx [hunk 19]
```diff
@@ -0,0 +1,74 @@
+import {Option} from "./option/Option";
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/select/Select.tsx:1]: absolute paths

File: mw-webapp/src/component/select/Select.tsx [hunk 20]
```diff
@@ -0,0 +1,74 @@
...
+import styles from "src/component/select/Select.module.scss";
+
+/**
+ * Option element for a select component
+ */
+export interface OptionType {
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/select/Select.tsx:7]: Why interface for options inside select component? Check please

File: mw-webapp/src/component/select/Select.tsx (hunk 15 above)
Comment (Ekaterina1994) [reply mw-webapp/src/component/select/Select.tsx:45]: Can you dont use properties selected and  defaultvalue?

File: mw-webapp/src/component/select/option/Option.tsx [hunk 21]
```diff
@@ -0,0 +1,16 @@
+
+interface OptionProps{
+  key: string;
+  value: string;
+  text: string;
+}
+export const Option: React.FC<OptionProps> = (props: OptionProps) => {
+  return(
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/select/option/Option.tsx:8]: add docs

Comment (Ekaterina1994) [inline mw-webapp/src/component/select/option/Option.tsx:8]: add docs to public members and properties

**Review by Ekaterina1994 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/component/select/Select.tsx (hunk 19 above)
Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx:1]: done

File: mw-webapp/src/component/select/Select.tsx (hunk 20 above)
Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx:7]: done

File: mw-webapp/src/component/select/option/Option.tsx (hunk 21 above)
Comment (Lelik7777) [reply mw-webapp/src/component/select/option/Option.tsx:8]: done

Comment (Lelik7777) [reply mw-webapp/src/component/select/option/Option.tsx:8]: done

File: mw-webapp/src/component/select/Select.tsx [hunk 22]
```diff
@@ -0,0 +1,53 @@
...
+   */
+  name: string;
+  /**
+   * Options list
+   */
+  options: OptionType[];
+  /**
+   * Default option
+   */
+  onChange: (e: React.ChangeEvent<HTMLSelectElement>) => void;
```
Comment (Howl404) [inline mw-webapp/src/component/select/Select.tsx:29]: Why there is "Default option", it's a callback

**Review by Howl404 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/component/select/Select.tsx (hunk 22 above)
Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx:29]: done

File: mw-webapp/src/component/select/option/Option.tsx [hunk 23]
```diff
@@ -0,0 +1,45 @@
...
+  /**
+   * Option`s value
+   */
+  value: string;
+  /**
+   * Inner option text
+   */
+  text: string;
+}
+export const Option: React.FC<OptionProps> = (props: OptionProps) => {
```
Comment (tritonJS826) [inline mw-webapp/src/component/select/option/Option.tsx:36]: All public members (exported variables, functions, types, etc) should be documented. Check everywhere in PR

**Review by tritonJS826 — CHANGES_REQUESTED** [review]
Almost done!)

File: mw-webapp/src/component/select/Select.tsx [hunk 24]
```diff
@@ -0,0 +1,53 @@
+import {Option} from "src/component/select/option/Option";
+import {OptionType} from "src/component/select/option/Option";
+import styles from "src/component/select/Select.module.scss";
+
+
+/**
+ * Select component
+ */
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/select/Select.tsx:8]: It's not a Select component/ It's Select props

File: mw-webapp/src/component/select/Select.tsx [hunk 25]
```diff
@@ -0,0 +1,53 @@
...
+export interface SelectProps {
+  /**
+   * Label's text
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/select/Select.tsx:11]: -> Select's label text

File: mw-webapp/src/component/select/Select.tsx [hunk 26]
```diff
@@ -0,0 +1,53 @@
...
+   */
+  label: string;
+  /**
+   * Default select value
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/select/Select.tsx:15]: -> Select's value

File: mw-webapp/src/component/select/option/Option.tsx [hunk 27]
```diff
@@ -0,0 +1,45 @@
+
+/**
+ * Option element for a select component
+ */
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/select/option/Option.tsx:4]: -> Option type

File: mw-webapp/src/component/select/option/Option.tsx [hunk 28]
```diff
@@ -0,0 +1,45 @@
...
+   */
+  value: string;
+  /**
+   * Option`s visible text
+   */
+  text: string;
+}
+/**
+* Option component
+*/
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/select/option/Option.tsx:21]: It's not component. 
-> Option props

**Review by Ekaterina1994 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/component/select/option/Option.tsx (hunk 23 above)
Comment (Lelik7777) [reply mw-webapp/src/component/select/option/Option.tsx:36]: got it

File: mw-webapp/src/component/select/Select.tsx [hunk 29]
```diff
@@ -0,0 +1,53 @@
+import {Option} from "src/component/select/option/Option";
+import {OptionType} from "src/component/select/option/Option";
+import styles from "src/component/select/Select.module.scss";
```
[the remark this answers is not in the document]
Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx:3]: done

File: mw-webapp/src/component/select/Select.tsx (hunk 26 above)
Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx:15]: done

File: mw-webapp/src/component/select/option/Option.tsx (hunk 28 above)
Comment (Lelik7777) [reply mw-webapp/src/component/select/option/Option.tsx:21]: done

**Comment by tritonJS826** [comment]
Almond st done! Please add docs for all public members. (All exported vars, funcs etc.)

**Review by Ekaterina1994 — APPROVED** [review]

**Review by tritonJS826 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/component/select/Select.tsx [hunk 30]
```diff
@@ -0,0 +1,52 @@
...
+   */
+  name: string;
+  /**
+   * Options list
+   */
+  options: OptionType[];
+  /**
+   * Callback triggered onChange select value
+   */
+  onChange: (e: React.ChangeEvent<HTMLSelectElement>) => void;
```
[the remark this answers is not in the document]
Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx:28]: done

File: mw-webapp/src/component/select/Select.tsx [hunk 31]
```diff
@@ -0,0 +1,54 @@
...
+   */
+  name: string;
+  /**
+   * Options list
+   */
+  options: OptionType[];
+  /**
+   * Callback triggered onChange select value
+   */
+  onChange: (value: React.ChangeEvent<HTMLSelectElement>) => void;
```
Comment (tritonJS826) [inline mw-webapp/src/component/select/Select.tsx:28]: now your interface is a lie. Did you check ant design select component? value is not an event

**Review by tritonJS826 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/component/select/Select.tsx (hunk 31 above)
Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx:28]: in art design use handler 
const onChange = (value: string) => {
  console.log(`selected ${value}`);
};
but if i use it throw errow:
Type '(value: string) => void' is not assignable to type 'ChangeEventHandler<HTMLSelectElement>'.
  Types of parameters 'value' and 'event' are incompatible.
    Type 'ChangeEvent<HTMLSelectElement>' is not assignable to type 'string'.

we need use in props of handler e:React.ChangeEvent<HTMLSelectElement>

Comment (Lelik7777) [reply mw-webapp/src/component/select/Select.tsx:28]: done

**Review by tritonJS826 — APPROVED** [review]

**Review by Howl404 — APPROVED** [review]

**Review by SergioAJS — APPROVED** [review]
